### PR TITLE
Identify and fix bugs

### PR DESCRIPTION
**Pull request by phated**
@jonschlinkert @doowb found some problems and fixed them.  Split the commits so you can see easier.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -53,7 +53,12 @@ module.exports = function(filepath) {
     });
 
     define(obj, 'dir', function() {
-      return path.dirname(filepath);
```
Comment (jonschlinkert) [inline index.js]: shouldn't we try to do the same thing as node here? Or is this similar to how node handles it?

File: .travis.yml [hunk 2]
```diff
@@ -2,11 +2,9 @@ sudo: false
 language: node_js
 node_js:
   - stable
-  - '5'
+  - '6'
```
Comment (jonschlinkert) [inline .travis.yml]: maybe add other recent versions of node as well?

File: test.js [hunk 3]
```diff
@@ -4,10 +4,18 @@ require('mocha');
...
 
 if (typeof path.parse === 'function') {
   describe('with native `path.parse` method:', function() {
+
+    var parsePath;
+
+    beforeEach(function() {
+      // Avoid cache
+      delete require.cache[require.resolve('./')];
+      parsePath = require('./');
```
Comment (jonschlinkert) [inline test.js]: good idea

**Review by jonschlinkert — COMMENTED** [review]
looks good, feel free to merge. I only have the questions about dirname and adding travis versions. But I'm good either way. thanks!

File: index.js (hunk 1 above)
Comment (phated) [reply index.js]: This logic matches the path.parse tests in this repo.  I suppose I can dig into their code to see the logic.

File: .travis.yml (hunk 2 above)
Comment (phated) [reply .travis.yml]: I'll switch it away from "stable" and use all evens like I did in the other repos.

File: index.js (hunk 1 above)
Comment (phated) [reply index.js]: They don't actually use other `path.*` methods within their impl and instead parse everything manually but this is the behavior exhibited so I think we are good.  The updates to the tests exposed this problem.

**Pull request merged**